graylog2-web-interface/src/components/sidecars/failure-tracking/SidecarFailureTrackingRows.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { Link } from 'components/common/router';
+import RelativeTime from 'components/common/RelativeTime';
+import Routes from 'routing/Routes';
+import StatusIndicator from 'components/sidecars/common/StatusIndicator';
+import SidecarStatusEnum from 'logic/sidecar/SidecarStatusEnum';
+import { Button } from 'components/bootstrap';
+
+import type { Collector, SidecarSummary } from '../types';
+
+const VerboseMessageContainer = styled.div`
+  height: 80px;
+  overflow-x: hidden;
+  overflow-y: scroll;
+  white-space: pre-wrap;
+  margin-bottom: 6px;
+`;
+
+const SecondaryText = styled.div`
+  color: #94979c;
+  font-style: italic;
+  font-size: 66%;
+`;
+
+const ShowDetailsLink = styled(Button)`
+  padding-left: 0;
+`;
+
+type Props = {
+  sidecar: SidecarSummary,
+  collectors: Collector[],
+  onShowDetails: (obj: { name: string, verbose_message: string }) => void,
+}
+
+const SidecarFailureTrackingRows = ({ sidecar, collectors, onShowDetails }: Props) => {
+  const annotation = sidecar.active ? '' : ' (inactive)';
+  let sidecarStatus = { status: null, message: null, id: null };
+
+  if (sidecar.node_details.status && SidecarStatusEnum.isValidStatusCode(sidecar.node_details.status.status)) {
+    sidecarStatus = {
```

Comment:
I am wondering if it would be nicer to use an effect hook here instead of a let. WDYT?